packages/neuron-ui/src/utils/hooks/useGetCountDownAndFeeRateStats.ts

```diff
@@ -0,0 +1,75 @@
+import { useState, useEffect, useCallback } from 'react'
+import { getFeeRateStats } from 'services/chain'
+import { AppActions, StateDispatch, useDispatch } from 'states'
+import { MEDIUM_FEE_RATE } from 'utils/const'
+
+type CountdownOptions = {
+  seconds?: number
+  interval?: number
+}
+
+const useGetCountDownAndFeeRateStats = ({ seconds = 30, interval = 1000 }: CountdownOptions = {}) => {
+  const [countDown, setCountDown] = useState(seconds)
+  const [timerId, setTimerId] = useState<NodeJS.Timeout | null>(null)
+  const [feeFatestatsData, setFeeFatestatsData] = useState<{
+    mean?: string
+    median?: string
+    suggestFeeRate: number
+  }>({ suggestFeeRate: MEDIUM_FEE_RATE })
+  const dispatch = useDispatch()
+
+  const handleGetFeeRateStatis = useCallback(
+    (stateDispatch: StateDispatch) => {
+      getFeeRateStats()
+        .then(res => {
+          const { mean, median } = res
+          const suggested = mean && median ? Math.max(1000, Number(mean), Number(median)) : MEDIUM_FEE_RATE
+
+          setFeeFatestatsData(states => ({ ...states, ...res, suggestFeeRate: suggested }))
+        })
+        .catch((err: Error) => {
+          stateDispatch({
+            type: AppActions.AddNotification,
+            payload: {
+              type: 'alert',
+              timestamp: +new Date(),
+              content: err.message,
+            },
+          })
+        })
+    },
+    [getFeeRateStats, setFeeFatestatsData]
+  )
+
+  useEffect(() => {
+    if (countDown >= 0) {
+      if (timerId) {
+        clearTimeout(timerId)
+      }
+
+      const newTimerId = setTimeout(() => {
+        setCountDown(count => (count <= 0 ? seconds : count - 1))
+      }, interval)
+
+      setTimerId(newTimerId)
+    } else {
+      setCountDown(seconds)
+    }
+
+    return () => {
+      if (timerId) {
+        clearTimeout(timerId)
+      }
+    }
+  }, [countDown, setCountDown, setTimerId])
```

Comment:
Every time the `count` changes, there will call setTimeout and clearTimeout, I think it's not good.

Reply:
Update with a setInterval.